### PR TITLE
Scan warning fixes v2

### PR DESCRIPTION
**Pull request by akallabeth**
Gets down the scanbuild warnings quite a bit.

```Bug Summary
Clang Version:	clang version 5.0.0-svn293191-1~exp1 (trunk)
Date:	Wed Feb 22 12:19:50 2017
Bug Summary
Bug Type	Quantity	Display?
All Bugs	165	
API	
Argument with 'nonnull' attribute passed null	3	
Dead store	
Dead assignment	149	
Dead increment	8	
Memory Error	
Memory leak	2	
Unix API	
Allocator sizeof operand mismatch	3	
```

### CODE REVIEW

**Comment by hardening** [comment]
Sorry to bother again but some of the astyle changes are very unfortunate regarding the limit of chars per lines. Some pieces of codes that were readable become ugly, perhaps we should just drop that limit from astyle, as the FreeRDP team members are very reasonable on that part and enforcing that just makes code harder to read...

**Comment by akallabeth** [comment]
@freerdp-bot test
@hardening I agree that it is not perfect everywhere but changing it again will only result in major reformatting of code already conforming to current settings.

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot come on test

**Comment by akallabeth** [comment]
@freerdp-bot not your day, hmm? test!

**Comment by akallabeth** [comment]
@freerdp-bot test

File: server/shadow/shadow_client.c [hunk 1]
```diff
@@ -796,17 +797,21 @@ static BOOL shadow_client_send_surface_bits(rdpShadowClient* client,
...
 	rdpUpdate* update;
-	rdpContext* context;
+	rdpContext* context = (rdpContext*) client;
 	rdpSettings* settings;
-	rdpShadowServer* server;
 	rdpShadowEncoder* encoder;
 	SURFACE_BITS_COMMAND cmd;
-	context = (rdpContext*) client;
+
+	if (!client || !context || !pSrcData)
```
Comment (mfleisz) [inline server/shadow/shadow_client.c]: Isn't client == context - checking one should be sufficient

File: server/shadow/shadow_client.c [hunk 2]
```diff
@@ -944,20 +949,25 @@ static BOOL shadow_client_send_bitmap_update(rdpShadowClient* client,
...
 	UINT32 maxUpdateSize;
 	UINT32 totalBitmapSize;
 	UINT32 updateSizeEstimate;
 	BITMAP_DATA* bitmapData;
 	BITMAP_UPDATE bitmapUpdate;
-	rdpShadowServer* server;
 	rdpShadowEncoder* encoder;
-	context = (rdpContext*) client;
+
+	if (!context || !client || !pSrcData)
```
Comment (mfleisz) [inline server/shadow/shadow_client.c]: again checking either client or context should be sufficient

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

File: libfreerdp/core/fastpath.c [hunk 3]
```diff
@@ -105,64 +112,108 @@ UINT16 fastpath_read_header(rdpFastPath* fastpath, wStream* s)
...
 	{
 		if (rdp->settings->EncryptionMethods == ENCRYPTION_METHOD_FIPS)
+		{
+			if (Stream_GetRemainingCapacity(s) < 4)
+				return FALSE;
+
 			Stream_Write(s, fpUpdatePduHeader->fipsInformation, 4);
+		}
+
+		if (Stream_GetRemainingCapacity(s) < 1)
```
Comment (nfedera) [inline libfreerdp/core/fastpath.c]: Shouldn't this be `< 8 ` ?

Comment (akallabeth) [inline libfreerdp/core/fastpath.c]: true

File: libfreerdp/core/gateway/rts.c [hunk 4]
```diff
@@ -438,46 +427,37 @@ int rts_send_CONN_A1_pdu(rdpRpc* rpc)
...
 	header.frag_length = 76;
 	header.Flags = RTS_FLAG_NONE;
 	header.NumberOfCommands = 4;
-
 	WLog_DBG(TAG, "Sending CONN/A1 RTS PDU");
-
-	VirtualConnectionCookie = (BYTE*) &(connection->Cookie);
-	OUTChannelCookie = (BYTE*) &(outChannel->Cookie);
+	VirtualConnectionCookie = (BYTE*) & (connection->Cookie);
+	OUTChannelCookie = (BYTE*) & (outChannel->Cookie);
```
Comment (nfedera) [inline libfreerdp/core/gateway/rts.c]: I don't understand these two changes above. Looks weird.

File: libfreerdp/core/gateway/rts.c [hunk 5]
```diff
@@ -494,37 +474,33 @@ int rts_send_CONN_B1_pdu(rdpRpc* rpc)
...
-
 	WLog_DBG(TAG, "Sending CONN/B1 RTS PDU");
-
-	VirtualConnectionCookie = (BYTE*) &(connection->Cookie);
-	INChannelCookie = (BYTE*) &(inChannel->Cookie);
-	AssociationGroupId = (BYTE*) &(connection->AssociationGroupId);
-
+	VirtualConnectionCookie = (BYTE*) & (connection->Cookie);
+	INChannelCookie = (BYTE*) & (inChannel->Cookie);
+	AssociationGroupId = (BYTE*) & (connection->AssociationGroupId);
```
Comment (nfedera) [inline libfreerdp/core/gateway/rts.c]: again ... weird formatting.

File: libfreerdp/core/gateway/rts.c [hunk 6]
```diff
@@ -794,67 +739,55 @@ int rts_command_length(rdpRpc* rpc, UINT32 CommandType, BYTE* buffer, UINT32 len
...
 	rts_pdu_header_init(&header);
 	header.frag_length = 56;
 	header.Flags = RTS_FLAG_OUT_CHANNEL;
 	header.NumberOfCommands = 3;
-
 	WLog_DBG(TAG, "Sending OUT_R2/A7 RTS PDU");
-
-	SuccessorChannelCookie = (BYTE*) &(nextOutChannel->Cookie);
-
+	SuccessorChannelCookie = (BYTE*) & (nextOutChannel->Cookie);
```
Comment (nfedera) [inline libfreerdp/core/gateway/rts.c]: please no space after the `&`

File: libfreerdp/core/message.c [hunk 7]
```diff
@@ -2272,26 +2618,20 @@ int update_message_queue_process_message(rdpUpdate* update, wMessage* message)
...
 	msgClass = GetMessageClass(message->id);
 	msgType = GetMessageType(message->id);
-
-	status = update_message_free_class(message, msgClass, msgType);
-
-	if (status < 0)
-		status = -1;
-
-	return 1;
+	return update_message_free_class(message, msgClass, msgType);
```
Comment (akallabeth) [inline libfreerdp/core/message.c]: @nfedera Yes, the function return value is currently not checked anywhere.

File: winpr/libwinpr/utils/wlog/UdpAppender.c [hunk 8]
```diff
@@ -209,14 +210,13 @@ wLogAppender* WLog_UdpAppender_New(wLog* log)
...
 		if (!appender->host)
-			goto error_host_alloc;
+			goto error_open;
 	}
 
 	return (wLogAppender*)appender;
 
 error_open:
 	free(appender->host);
-error_host_alloc:
```
Comment (nfedera) [inline winpr/libwinpr/utils/wlog/UdpAppender.c]: I'd opt for keeping this label

**Comment by akallabeth** [comment]
@freerdp-bot test

File: channels/rail/client/rail_orders.c [hunk 9]
```diff
@@ -120,8 +119,8 @@ UINT rail_read_server_exec_result_order(wStream* s, RAIL_EXEC_RESULT_ORDER* exec
 	Stream_Read_UINT16(s, execResult->execResult); /* execResult (2 bytes) */
 	Stream_Read_UINT32(s, execResult->rawResult); /* rawResult (4 bytes) */
 	Stream_Seek_UINT16(s); /* padding (2 bytes) */
-
-	return rail_read_unicode_string(s, &execResult->exeOrFile) ? CHANNEL_RC_OK : ERROR_INTERNAL_ERROR; /* exeOrFile */
+	return rail_read_unicode_string(s,
+	                                &execResult->exeOrFile) ? CHANNEL_RC_OK : ERROR_INTERNAL_ERROR; /* exeOrFile */
 }
```
Comment (hardening) [inline channels/rail/client/rail_orders.c]: oh thank you astyle..

Comment (bmiklautz) [inline channels/rail/client/rail_orders.c]: yeah this looks really crappy ;(

**Pull request merged**